src/jenkins.py

```diff
@@ -47,17 +78,29 @@ def wait_jenkins_ready(timeout: int = 140, check_interval: int = 10) -> None:
     start_time = now = datetime.now()
     min_wait_seconds = timedelta(seconds=timeout)
     while now - start_time < min_wait_seconds:
-        if _is_jenkins_ready():
+        if _is_ready():
             break
         now = datetime.now()
         sleep(check_interval)
     else:
-        if _is_jenkins_ready():
+        if _is_ready():
             return
         raise TimeoutError("Timed out waiting for Jenkins to become ready.")
 
 
-def get_admin_credentials(connectable_container: Container) -> Credentials:
+class Credentials(typing.NamedTuple):
```

Comment:
I think a `@dataclass` might be better for this, as it's not really a tuple.

Reply:
May I know why a `@dataclass` would be more suitable here? My understanding is that `tuple`s are immutable by default and that's what I would want when passing relevant group of data around. a `@dataclass` with `frozen=True` could do the same. Would you be able to help me understand what you mean by "as it's not really a tuple" please?
Thank you for the review!

Reply:
Yeah, sorry, should have stated my motivation. I used to use namedtuple (the old-school one) all the time, so it's not *bad*, but the reason I consider data classes better for this kind of structure is that tuples are [sequence types](https://docs.python.org/3/library/stdtypes.html#typesseq), so are indexable by integer (but `credentials[0]` and `credentials[1]` doesn't really makes) and are "accidentally equal" to other tuples of the same length. See the ["Why not just use a namedtuple?"](https://peps.python.org/pep-0557/#why-not-just-use-namedtuple) section in PEP 557, the dataclasses proposal, for more info.

Reply:
BTW, overall I think this PR is good stuff! I'm just thinking we should try to land on a few style things like "use dataclasses where appropriate" early on.

CC @jdkandersson: assuming you agree, maybe when you make updates to ISD014 you can show the State example with a dataclass too?

Reply:
This feedback was extremely helpful, I'll share it with the team as well. Thank you!

Reply:
Yeh I think dataclasses are also good, the spec should note that I agree

Reply:
Fixed, thank you!